api/cloud/oc_cloud_manager.c

```diff
@@ -587,10 +613,15 @@ cloud_manager_login_handler(oc_client_response_t *data)
   oc_cloud_context_t *ctx = (oc_cloud_context_t *)data->user_data;
   oc_remove_delayed_callback(ctx, cloud_manager_login_async);
   bool handleUnauthorizedByRefresh = cloud_context_has_refresh_token(ctx);
-  bool retry = false;
-  oc_cloud_error_t ret = _login_handler(ctx, data, /*retryIsActive*/ true,
-                                        !handleUnauthorizedByRefresh);
-  if (ret == CLOUD_OK) {
+  oc_cps_t cps = OC_CPS_FAILED;
+  oc_cloud_error_t err = _login_handler(ctx, data, &cps);
+  if ((ctx->store.status & OC_CLOUD_REGISTERED) == 0) {
```

Comment:
`_login_handler` no longer calls `cloud_context_clear` so there's no code path that removes the `OC_CLOUD_REGISTERED` flag. so this check canbe removed or moved before the call to `_login_handler` if we want to keep it

Reply:
But the handler could be called in with `ctx->store.status & OC_CLOUD_REGISTERED) == 0` so this is why is this there.